notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/NotificationEventUtils.java

```diff
@@ -91,13 +114,25 @@ public static AvroKey toAvroKeyRecord(Stream stream) {
 
     val name = stream.getKey().getName();
     val version = stream.getKey().getVersion();
+    val domainName = stream.getKey().getDomain();
 
-    return AvroKey.newBuilder()
-        .setId(name)
-        .setVersion(version)
+    var domainKey = AvroKey.newBuilder()
+        .setId(domainName)
+        .setType(AvroKeyType.DOMAIN)
         .setParent(null)
```

Comment:
Again do we need to set null?

Reply:
Yes, because avro schema doesn't have a default value. It will cause an error if it is not set explicitly.
```
    union {null, AvroKey} parent;
```

Reply:
A default value could set explicitly though. But I'm preserving the original schema design.
```
union {null, AvroKey} parent = null; 
```

Reply:
In that case I don't really see how a builder is the appropriate construction mechanism.

can we just have an all args constructor and be done with it?

Reply:
@nahguam do you see any trouble setting a default null for the parent?:
```
union {null, AvroKey} parent = null; 
```